packages/jira-adapter/src/filters/automation/handle_template_expressions.ts

```diff
@@ -0,0 +1,142 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  Change, Element, getChangeData, InstanceElement, isInstanceElement,
+  ReferenceExpression, TemplateExpression, TemplatePart, Values,
+} from '@salto-io/adapter-api'
+import { extractTemplate, replaceTemplatesWithValues, resolveTemplates } from '@salto-io/adapter-utils'
+import { logger } from '@salto-io/logging'
+import { collections } from '@salto-io/lowerdash'
+import _ from 'lodash'
+import { FilterCreator } from '../../filter'
+
+const { awu } = collections.asynciterable
+const log = logger(module)
+const REFERENCE_MARKER_REGEX = /(\{\{)(.+?)(}})/
+
+
+type Component = Record<string, Values>
+const simpleIssueRegex = new RegExp(/issue\.[a-zA-Z ]+(?: |$)/, 'g')
+const issueFieldNameToTypeMapping: Record<string, string> = {
+  key: 'Key',
+  project: 'Project',
+  status: 'Status',
+  summary: 'Summary',
+  versions: 'Affects versions',
+  resolution: 'Resolution',
+  updated: 'Updated',
+  'Story Points': 'Story Points',
+  'Story Points estimate': 'Story Points estimate',
+  created: 'Created',
+  duedate: 'Due Date',
+}
+
+const issueTypeNameToFieldMapping = Object.fromEntries(
+  Object.entries(issueFieldNameToTypeMapping).map(([k, v]) => [v, k])
+)
+
+const getAutomations = (instances: InstanceElement[]): InstanceElement[] =>
+  instances.filter(instance => instance.elemID.typeName === 'Automation')
+
+// This function receives a string that contains issue references and replaces
+// it with salto style templates.
+const stringToTemplate = (referenceSt: string,
+  fieldsByName: Record<string, InstanceElement>): TemplateExpression | string => {
+  const handleJiraReference = (expression: string, ref: RegExpMatchArray): TemplatePart => {
+    const referenceArr = (ref.pop() ?? '').split('.')
+    if (referenceArr.length === 2) {
+      const elem = fieldsByName[issueFieldNameToTypeMapping[referenceArr[1]] ?? referenceArr[1]]
+      if (elem) {
+        return new ReferenceExpression(elem.elemID, elem)
+      }
+    }
+    // if no id was detected we return the original expression.
+    return expression
+  }
+  return extractTemplate(referenceSt,
+    [REFERENCE_MARKER_REGEX],
+    expression => {
+      const jiraReference = expression.match(simpleIssueRegex)
+      if (jiraReference) {
+        return handleJiraReference(expression, jiraReference)
+      }
+      return expression
+    })
+}
+
+const replaceFormulasWithTemplates = async (instances: InstanceElement[]): Promise<void> => {
+  try {
+    getAutomations(instances).forEach(instance => {
+      const fieldsByName = _.keyBy(instances.filter(i => i.elemID.typeName === 'Field'
+        && _.isString(i.value.name)), (i: InstanceElement): string =>
+        i.value.name);
```

Comment:
1. readability
```suggestion
  const fieldsByName = _.keyBy(
    instances.filter(i => i.elemID.typeName === 'Field' && _.isString(i.value.name)),
    (i: InstanceElement): string => i.value.name
  );
```
2. I don't think we can assume fields are unique by their name in jira... do we have other ways to filter these to make sure we are adding a reference to the right one? @alonstern please chime in if you have more context on this :)

Reply:
2. Apparently it's possible. Also from trying it and fetching it looks like the current adapter doesn't really handle this scenario. I'm not sure what the right behavior is.